src/PrimaryAuthController.js

```diff
@@ -141,9 +142,14 @@ function (Okta, PrimaryAuthForm, SocialAuth, PrimaryAuthModel, Util, BaseLoginCo
 
       // Add SocialAuth view only when the idps are configured. If configured, 'socialAuthPositionTop'
       // will determine the order in which the social auth and primary auth are shown on the screen.
+      var addButtonSettings = this.model.settings.get('features.addButton');
+
       if (options.settings.get('socialAuthConfigured')) {
         this.add(SocialAuth, {prepend: options.settings.get('socialAuthPositionTop')});
       }
+      if (addButtonSettings && addButtonSettings.visibleWhen) {
```

Comment:
we need to evaluate if ```visibleWhen``` is a function or not. If it is, we need to call that function.